### DISCUSSION TITLE
Release 0.22

### DISCUSSION THREAD

**Discussion by viktormalik** [bpftrace/bpftrace#3560] [Development]
It's been a while since we did the last release and we have a lot of unreleased stuff already.

Let's discuss here what needs to be included in the release.

As discussed some while ago, we may want to consider crafting a release branch before the actual release to enable pulling new features into master while not blocking the release.

**Comment by viktormalik** [comment]
At the moment, we have one issue tagged into v0.22:
- https://github.com/bpftrace/bpftrace/issues/3498

From the other issues, these should be considered:
- https://github.com/bpftrace/bpftrace/issues/3557
- https://github.com/bpftrace/bpftrace/issues/3543
- https://github.com/bpftrace/bpftrace/issues/3540

From the open PRs, we should IMO wait for:
- https://github.com/bpftrace/bpftrace/pull/3550
- https://github.com/bpftrace/bpftrace/pull/3433

LLVM 19 support is tricky as Nix will only have LLVM 19 in 24.11 which is planned for Nov 22 (https://github.com/NixOS/nixpkgs/issues/339153). To speed things up, we could switch to unstable for a while just to get the testing coverage for LLVM 19.

**Reply by hhoffstaette** [reply]
> * [cmake: Bump max LLVM version to 19 #3433](https://github.com/bpftrace/bpftrace/pull/3433)

If it helps: I added support for LLVM 19 to my Gentoo build of bpftrace-0.21.2 and it works without a hitch. Other than CI I don't see why this shouldn't be enabled, since distros are still free to use -18.

**Comment by viktormalik** [comment]
Also we need https://github.com/bpftrace/bpftrace/pull/3561 to make sure that a released version depends on a released libbpf.

**Comment by jordalgo** [comment]
@viktormalik Thanks for getting this going.

> we may want to consider crafting a release branch 

+1 for this.

> - https://github.com/bpftrace/bpftrace/issues/3540

I fixed half of this issue with: https://github.com/bpftrace/bpftrace/pull/3542. I can try to look into the other part soon.

> - https://github.com/bpftrace/bpftrace/pull/3550

This is merged now though TBH I'm slightly nervous I didn't think of or catch all the weird edge cases around block scoping and variable ordering. This is another reason I'm pro "release branch".

> - https://github.com/bpftrace/bpftrace/pull/3433

I'm a little unclear why we want to wait for this for a release but I'm a little out of my depth.

**Reply by viktormalik** [reply]
Sorry, I just noticed that I never replied to this.

> > * [cmake: Bump max LLVM version to 19 #3433](https://github.com/bpftrace/bpftrace/pull/3433)
> 
> I'm a little unclear why we want to wait for this for a release but I'm a little out of my depth.

B/c many rolling distros (like Fedora) have LLVM 19 in their latest releases so we definitely want the new bpftrace release support LLVM 19. It would just be nice to also get test coverage from CI and for that we need new nixpkgs release to happen (which should come in the following days).

**Comment by viktormalik** [comment]
I believe we now have everything ready for the release.

**Reply by jordalgo** [reply]
Awesome. Are you going to cut the release? I also volunteer if you're busy

**Reply by viktormalik** [reply]
As I wrote in the IRC channel, I can cut the release but not sure if this is the best time of the year. I'm off from the next Saturday and would not like to end up with a broken release for ~3 weeks if nobody has time to fix things during Christmas :grin:

**Reply by jordalgo** [reply]
I'm fine to wait till the new year. That being said I'm around to fix stuff. @ajor is on PTO - not sure about @danobi
